File: csharp/main/LiteralType.cs
using System;

namespace Deltix.Luminary
{
	public sealed class LiteralType : Literal, ILiteral<Type>
	{
		public LiteralType(Type value)
			: base(LiteralKind.Type)
		{
			Value = value;
		}

		public Type Value { get; }

		public override String ToString()
		{
			return $"typeof({Value})";
		}
	}
}

File: java/src/main/java/deltix/luminary/LiteralType.java
package deltix.luminary;

public class LiteralType extends Literal {
    private final Type value;

    public LiteralType(Type value) {
        super(LiteralKind.TYPE);
        this.value = value;
    }

    public Type getValue() {
        return value;
    }

    @Override
    public String toString() {
        return "typeof(" + value + ")";
    }
}
